torchreid/data/datasets/image/classification.py

```diff
@@ -9,88 +9,94 @@ class Classification(ImageDataset):
     """Classification dataset.
     """
 
-    def __init__(self, root='', dataset_id=0, load_masks=False, cl_data_dir='cl', cl_version='v1', **kwargs):
+    def __init__(self, root='', mode='train', dataset_id=0, load_masks=False, **kwargs):
         if load_masks:
             raise NotImplementedError
 
         self.root = osp.abspath(osp.expanduser(root))
-        self.dataset_dir = osp.join(self.root, f'{cl_data_dir}_{cl_version}')
-        self.data_dir = self.dataset_dir
-
-        self.images_dir = osp.join(self.data_dir, 'images')
-        self.train_annot = osp.join(self.data_dir, 'train.txt')
-        self.test_annot = osp.join(self.data_dir, 'val.txt')
+        self.data_dir = osp.dirname(self.root)
```

Comment:
`self.load_annotation` method expects full path to the folder with images like `osp.join(osp.dirname(s), 'images')`.

Reply:
If annotation is supposed to store paths relative to root, that works: `images/img1.jpg 10`.
In this case we don't need to hardcode or pass additionally the images folder.

Reply:
Agree, but it will require to change all existing annotation files. From other hand, we don't have any reid model in OTE, so it should not impact them but the changes should be done as soon as reid models will be ready for model templates (faces, vehicles).

Reply:
Nikita claims that the reference here is the copy imagenet val in AC validation folder on the share. It contains relative to root paths:
```
./ILSVRC2012_val_00000001.JPEG 65
```
But who knows actually.

Reply:
These changes won't affect reid models unless they use classification datasets. Face and person models use unique-named datasets with custom handlers included to this repo, so we can use them as before.
And I think reid & FR are very far from converting into templates. At least we should first choose annotation format and a way how CVAT will handle these tasks. As far as I remember DS was using semi-manual pipeline to finally obtain a market-like annotation.